File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) < 3:
            return max(nums)
        prev = nums[2] + nums[0]
        twoPrev = nums[1]
        threePrev = nums[0]
        for i in range(3, len(nums)):
            temp = nums[i] + max(twoPrev, threePrev)
            threePrev = twoPrev
            twoPrev = prev
            prev = temp
        return max(prev, twoPrev)

File: Data Structures & Algorithms/jump-game-ii/submission-0.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int res = 0;
        int r = 0, l = 0;
        

        while (r < nums.size() - 1) {
            int maxJump = 0;
            for (int i = l; i <= r; i++) {
                maxJump = max(maxJump, i + nums[i]);
            }
            l = r + 1;
            r = maxJump;
            res += 1;
        }

        return res;
    }
};
